CRM/Civioffice/DocumentRendererType.php

```diff
@@ -100,45 +112,28 @@ public abstract function render(
     ): array;
 
     /**
-     * resolve all tokens
-     *
-     * @param array $token_names
-     *   the list of all token names to be replaced
+     * Adds implicit token contexts and builds the corresponding TokenProcessor context schema.
      *
-     * @param integer $entity_id
-     *   entity ID, e.g. contact_id
-     *
-     * @param string $entity_type
-     *   entity type, e.g. 'contact'
-     *
-     * @return array
-     *   list of token_name => token value
-     */
-    public function resolveTokens($token_names, $entity_id, $entity_type = 'contact'): array
-    {
-        // TODO: implement
-        // TODO: use additional token system
-        throw new Exception('resolveTokens not implemented');
-    }
-
-    /**
-     * Replace all tokens with {token_name} and {$smarty_var.attribute} format
-     *
-     * @param $string
-     *   A string including tokens to be replaced.
-     * @param string $entity_type
-     *   The entity type, e.g. 'contact'. The entity type "civioffice" refers to tokens defined by CiviOffice itself.
      * @param array $token_contexts
-     *   A list of token contexts, keyed by token entity (e.g. "contact" or "civioffice").
+     *   An array of token contexts, with entity types as keys and an array with context as values, e. g.
+     *   [
+     *     'contribution' => ['entity_id' => 123],
+     *   ]
+     *   This will be extended with the corresponding contact, and anything else that can be implicitly derived from the
+     *   entity given.
      *
-     * @return string
-     *   The input string with the tokens replaced.
+     * @return array
```

Comment:
Does not match method signature.

Reply:
Fixed with 5ee3fba9a0f95f9bcc5130956832ad5f818ccdfb.